stubs/docker/docker/utils/utils.pyi

```diff
@@ -1,34 +1,67 @@
-def version_lt(v1, v2): ...
-def version_gte(v1, v2): ...
-def convert_port_bindings(port_bindings): ...
-def convert_volume_binds(binds): ...
-def convert_tmpfs_mounts(tmpfs): ...
-def convert_service_networks(networks): ...
-def parse_repository_tag(repo_name): ...
-def parse_host(addr, is_win32: bool = False, tls: bool = False): ...
-def parse_devices(devices): ...
-def kwargs_from_env(environment: Incomplete | None = None): ...
-def convert_filters(filters): ...
-def datetime_to_timestamp(dt): ...
-def parse_bytes(s): ...
+def create_ipam_pool(*args, **kwargs) -> NoReturn: ...
+def create_ipam_config(*args, **kwargs) -> NoReturn: ...
+def decode_json_header(header: str | ReadableBuffer): ...
+def compare_version(v1: str, v2: str) -> Literal[0, -1, 1]: ...
+def version_lt(v1: str, v2: str) -> bool: ...
+def version_gte(v1: str, v2: str) -> bool: ...
+def convert_port_bindings(
+    port_bindings: dict[object, Incomplete],  # we are converts key to a string using str()
+) -> dict[str, list[dict[str, str]]]: ...
+@overload
+def convert_volume_binds(binds: list[_T]) -> list[_T]: ...
+@overload
+def convert_volume_binds(binds: dict[str | bytes, Incomplete]) -> list[str]: ...
+@overload
+def convert_tmpfs_mounts(tmpfs: dict[_K, _V]) -> dict[_K, _V]: ...
+@overload
+def convert_tmpfs_mounts(tmpfs: list[str]) -> dict[str, str]: ...
+@overload
+def convert_service_networks(networks: None) -> None: ...
+@overload
+def convert_service_networks(networks: list[str | _T]) -> list[_T]: ...
+def parse_repository_tag(repo_name: str) -> tuple[str, str | None]: ...
+@overload
+def parse_host(addr: None, is_win32: Literal[True], tls: bool = False) -> Literal["npipe:////./pipe/docker_engine"]: ...
+@overload
+def parse_host(addr: None, is_win32: Literal[False], tls: bool = False) -> Literal["http+unix:///var/run/docker.sock"]: ...
+@overload
+def parse_host(addr: str | None, is_win32: bool = False, tls: bool = False) -> str | bytes: ...
+def parse_devices(devices: Iterable[str | dict[str, Incomplete]]) -> list[dict[str, Incomplete]]: ...
+def kwargs_from_env(environment: Mapping[str, Incomplete] | None = None) -> dict[Literal["base_url", "tls"], str | TLSConfig]: ...
+def convert_filters(filters) -> str: ...
+def datetime_to_timestamp(dt: datetime.datetime) -> int: ...
+def parse_bytes(s: float | str) -> float: ...
 def normalize_links(links): ...
-def parse_env_file(env_file): ...
-def split_command(command): ...
-def format_environment(environment): ...
-def format_extra_hosts(extra_hosts, task: bool = False): ...
-def create_host_config(self, *args, **kwargs) -> None: ...
+def parse_env_file(env_file: FileDescriptorOrPath) -> dict[str, str]: ...
+def split_command(command: str | _ShlexInstream) -> list[str]: ...
+def format_environment(environment: dict[str, object | None]) -> list[str]: ...
+def format_extra_hosts(
+    extra_hosts: dict[object, object], task: bool = False  # we are converts key and value to a string using str()
```

Comment:
`dict[str, str]` is not a valid `dict[object, object]` since `dict` is invariant
```suggestion
    extra_hosts: Mapping[object, object], task: bool = False  # keys and values are converted to str
```